core/templates/dev/head/galleries/Gallery.js

```diff
@@ -203,6 +203,125 @@ oppia.controller('Gallery', [
       $rootScope.loadingMessage = '';
     });
 
+    // Below is the width of each tile
+    var TILEDISPLAYWIDTH = 211.953;
+    // If the value below is changed, the corresponding CSS values must be
+    // changed in oppia.css.
+    var MAXTILEDISPLAY = 4;
+    var maxGalleryTileWidth = TILEDISPLAYWIDTH * MAXTILEDISPLAY;
+    $scope.tileDisplayCount = 0;
+    $scope.galleryTilesResizedWidth = 0;
+
+    var calculateCarouselCssWidth = function() {
+      var windowWidth = $(window).width() * 0.85;
+      $scope.tileDisplayCount = Math.floor(windowWidth / TILEDISPLAYWIDTH);
+      $scope.galleryTilesResizedWidth = $scope.tileDisplayCount *
+        TILEDISPLAYWIDTH;
+      if ($scope.galleryTilesResizedWidth > maxGalleryTileWidth) {
+        $scope.galleryTilesResizedWidth = maxGalleryTileWidth;
+        $scope.tileDisplayCount = MAXTILEDISPLAY;
+      }
+      $('.oppia-gallery-tiles-carousel').css({
+        width: $scope.galleryTilesResizedWidth + 'px'
+      });
+
+      // The following determines whether to enable left scroll after resize
+      for (i = 0; i < $('.oppia-gallery-tiles-carousel').length; i++) {
+        var carouselJQuerySelect = '.oppia-gallery-tiles:eq(n)'.replace(
+          'n', i);
+        var carouselScroll = $(carouselJQuerySelect).scrollLeft();
```

Comment:
I don't think I understand what this variable, or the following one, represents. Could you please clarify the names? (Are they something like carouselOffsetPx and leftmostCardIndex?)